.buildkite/steps/build-lambda.sh

```diff
@@ -3,6 +3,11 @@ set -eu
 
 make handler.zip
 
-# set a version for later steps
-buildkite-agent meta-data set version \
-  "$(awk -F\" '/const Version/ {print $2}' version/version.go)"
+if [[ -z "${BUILDKITE_TAG:-}" ]]; then
+  VERSION="$BUILDKITE_COMMIT"
+else
+  VERSION=$(awk -F\" '/const Version/ {print $2}' version/version.go)
+fi
+
+# set version for later steps
+buildkite-agent meta-data set version "$VERSION"
```

Comment:
have we ensured that SAM will take a version that's not semver? i remember something about the field on the cfn template mentioning semver specifically, but i might be mistaken